packages/modular-scripts/src/utils/config.ts

```diff
@@ -0,0 +1,91 @@
+import { cosmiconfig } from 'cosmiconfig';
+import path from 'path';
+import getModularRoot from './getModularRoot';
+import * as logger from './logger';
+
+// Where cosmiconfig can look for the configuration
+const searchPlaces = [
+  '.modular.js',
+  'package.json',
+  `.modularrc`,
+  `.modularrc.json`,
+  `.modularrc.yaml`,
+  `.modularrc.yml`,
+  `.modularrc.js`,
+  `.modularrc.cjs`,
+  `modular.config.js`,
+  `modular.config.cjs`,
+];
+
+// Look for configuration file
+const modularRoot = getModularRoot();
+const explorer = cosmiconfig('modular', { searchPlaces });
+const configuration = explorer.search(path.join(modularRoot, 'package.json'));
+
+// Interface with all configurations
+interface ConfigObject {
+  useModularEsbuild: boolean;
+}
+
+type ConfigObjectKey = keyof ConfigObject;
+
+/**
+ * Get the configured value for a given configuration field.
+ * Rejects if no configuration file is present or the queired field is not present.
+ * @param configEntry Field containing the configuration variable to read
+ * @returns Value of configuration field queried if present
+ */
+export async function getConfiguration(
+  configEntry: ConfigObjectKey,
+): Promise<boolean> {
+  const loadedConfiguration = await configuration;
+  // Handle no or empty configuration - debug log? don't think we should error should we?
+  if (loadedConfiguration) {
+    // Error if configuration doesn't match our interface?
+    const config = loadedConfiguration.config as ConfigObject;
+    const value = config[configEntry];
+    if (value) {
+      return value;
+    } else {
+      throw new Error(
```

Comment:
I think users can always omit config entries (that's why we need defaults, see comment below)